oversweet-api/src/main/java/com/depromeet/oversweet/drink/service/DrinkStatisticsService.java

```diff
@@ -0,0 +1,88 @@
+import com.depromeet.oversweet.domain.record.repository.FindDailyRecordsRepository;
+import com.depromeet.oversweet.drink.dto.DrinkDailyDetailInfo;
+import com.depromeet.oversweet.drink.dto.DrinkDailySugarStatisticsResponse;
+import com.depromeet.oversweet.drink.dto.DrinkDailySugarTotalStatisticsInfo;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class DrinkStatisticsService {
+
+    private final FindDailyRecordsRepository findDailyRecordsRepository;
+    private final FindMemberRepository findMemberRepository;
+
+    public DrinkDailySugarStatisticsResponse retrieveUserDailySugarStatistics(final Long memberId) {
+        // TODO 유저 검증
+
+        // 유저의 하루 적정 섭취량 가져오기
+        final MemberEntity findMember = findMemberRepository.findById(memberId);
+
+        // 오늘 (데일리 날짜 확인) 00:00 ~ 23:59
+        final LocalDateTime nowDateTime = LocalDateTime.now(ZoneId.of("Asia/Seoul"));
+        final LocalDateTime startDateTime = nowDateTime.with(LocalTime.MIN);
+        final LocalDateTime endDateTime = nowDateTime.with(LocalTime.MAX);
+
+        // 해당 유저의 오늘(데일리) 섭취 음료 조회 Repository
+        final List<RecordEntity> dailyRecords = findDailyRecordsRepository.findDailyRecordsByLocalDatetime(memberId, startDateTime, endDateTime);
+
+        // 조회 한 데이터 가공 하기. ( 당 섭취량 계산, 칼로리 계산 )
+        final int totalIntakeSugar = getTotalIntakeSugar(dailyRecords);
+        final int totalCalorie = getTotalCalorie(dailyRecords);
+        final int remainingSugar = getRemainingSugar(dailyRecords, findMember.getDailySugar());
+
+        // Response로 만들어 주기
+        final DrinkDailySugarTotalStatisticsInfo totalStatisticsInfo =
+                DrinkDailySugarTotalStatisticsInfo.of(findMember.getDailySugar(), remainingSugar, totalIntakeSugar, totalCalorie);
+
+        final List<DrinkDailyDetailInfo> dailyDetailInfo = getDailyDrinks(dailyRecords);
+
+        // return
+       return DrinkDailySugarStatisticsResponse
+                .builder()
+                .totalDailyStatistics(totalStatisticsInfo)
+                .dailyDrinks(dailyDetailInfo)
+                .build();
+    }
+
+    private List<DrinkDailyDetailInfo> getDailyDrinks(final List<RecordEntity> dailyRecords) {
+        return dailyRecords.stream()
+                .map(dr -> {
+                    return DrinkDailyDetailInfo.of(dr.getDrink());
+                }).toList();
+    }
+
+    private int getTotalIntakeSugar(final List<RecordEntity> recordEntities) {
+        return recordEntities.stream()
+                .mapToInt(RecordEntity::getIntakeSugar)
+                .sum();
+    }
+
+    private int getTotalCalorie(final List<RecordEntity> dailyRecords) {
+        return dailyRecords.stream()
+                .mapToInt(record -> record.getDrink().getCalorie())
+                .sum();
+    }
```

Comment:
해당 코드에서는 몇잔 (Record.count) 필드를 고려하지 못 한거 같아요
하나의 Record row 기준, 섭취한 총 칼로리를 계산하기 위해서는 count 필드 (몇잔)을 고려할 필요가 있어요

```suggestion
    private int getTotalCalorie(final List<RecordEntity> dailyRecords) {
        return dailyRecords.stream()
                .mapToInt(record -> record.totalCalorie())
                .sum();
    }
```

Record Entity에는 아래와 같은 method가 있으면 좋을 것 같아요~
```java
    private int totalCalorie() {
        return this.count * this.drink.getCalorie();
    }
```

밑의 코멘트에서 언급을 했지만, **만약 일급 컬렉션으로 감싸면 해당 컬렉션 객체에게 역할을 위임할 수 있을 것 같아요~**